src/Framework/Framework/Binding/DotvvmCapabilityProperty.cs

```diff
@@ -338,20 +338,17 @@ internal static IControlAttributeDescriptor InitializeArgument(ICustomAttributeP
                     
                     var isNullable = propertyType.IsNullable() || propertyType.UnwrapValueOrBinding().IsNullable();
                     if (!defaultValue.HasValue && !isNullable)
-                        dotvvmProperty.MarkupOptions.Required = true;
+                        dotvvmProperty.MarkupOptions._required ??= true;
 
                     if (typeof(IBinding).IsAssignableFrom(propertyType))
-                        dotvvmProperty.MarkupOptions.AllowHardCodedValue = false;
+                        dotvvmProperty.MarkupOptions._allowHardCodedValue ??= false;
                     else if (!typeof(ValueOrBinding).IsAssignableFrom(propertyType.UnwrapNullableType()))
-                        dotvvmProperty.MarkupOptions.AllowBinding = false;
-
-                    if (!attributeProvider.IsDefined(typeof(MarkupOptionsAttribute), true))
-                    {
-                        if (typeof(IDotvvmObjectLike).IsAssignableFrom(type) ||
-                            typeof(ITemplate).IsAssignableFrom(type) ||
-                            typeof(IEnumerable<IDotvvmObjectLike>).IsAssignableFrom(type))
-                            dotvvmProperty.MarkupOptions.MappingMode = MappingMode.InnerElement;
-                    }
+                        dotvvmProperty.MarkupOptions._allowBinding ??= false;
+
+                    if (typeof(IDotvvmObjectLike).IsAssignableFrom(type) ||
+                        typeof(ITemplate).IsAssignableFrom(type) ||
+                        typeof(IEnumerable<IDotvvmObjectLike>).IsAssignableFrom(type))
+                        dotvvmProperty.MarkupOptions._mappingMode = MappingMode.InnerElement;
```

Comment:
Shall we overwrite `mappingMode` here, or set it only when it's null via `??=`? 
What if someone sets it to `Exclude`?